packages/flutter/lib/src/material/stepper.dart

```diff
@@ -0,0 +1,648 @@
+// Copyright 2016 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'package:flutter/widgets.dart';
+import 'package:meta/meta.dart';
+
+import 'button.dart';
+import 'colors.dart';
+import 'debug.dart';
+import 'flat_button.dart';
+import 'icon.dart';
+import 'icons.dart';
+import 'ink_well.dart';
+import 'material.dart';
+import 'theme.dart';
+
+// TODO(dragostis): Missing functionality:
+//   * mobile horizontal mode w/ adding/removing steps
+//   * alternative labeling
+//   * stepper feedback in the case of high-latency interactions
+
+/// The state of a step which is used to control the style of the circle and
+/// text.
+///
+/// See also:
+///
+///  * [Step]
+enum StepState {
+  /// A step that displays its index in its circle.
+  indexed,
+  /// A step that displays a pencil icon in its circle.
+  editing,
+  /// A step that displays a tick icon in its circle.
+  complete,
+  /// A step that is disabled and does to react to taps.
+  disabled,
+  /// A step that is currently having an error. e.g. the use has submitted wrong
+  /// input.
+  error
+}
+
+/// The type of the [Stepper] that determines it's slyle.
+enum StepperType {
+  /// A vertical layout of the steps with their content in-between the titles.
+  vertical,
+  /// A horizontal layout of the steps with their content below the titles.
+  horizontal
+}
+
+const TextStyle _kStepStyle = const TextStyle(
+  fontSize: 12.0,
+  color: Colors.white
+);
+const TextStyle _kActiveTitleStyle = const TextStyle(
+  fontSize: 14.0,
+  fontWeight: FontWeight.w500,
+  color: Colors.black87
+);
+const TextStyle _kActiveSubtitleStyle = const TextStyle(
+  fontSize: 12.0,
+  color: Colors.black54
+);
+const TextStyle _kDisabledTitleStyle = const TextStyle(
+  fontSize: 14.0,
+  fontWeight: FontWeight.w500,
+  color: Colors.black38
+);
+const TextStyle _kDisabledSubtitleStyle = const TextStyle(
+  fontSize: 12.0,
+  color: Colors.black38
+);
+final TextStyle _kErrorTitleStyle = new TextStyle(
+  fontSize: 14.0,
+  fontWeight: FontWeight.w500,
+  color: Colors.red[500]
+);
+final TextStyle _kErrorSubtitleStyle = new TextStyle(
+  fontSize: 12.0,
+  color: Colors.red[500]
+);
+
+/// A material step used in [Stepper]. The step can have a title and subtitle,
+/// an icon within its circle, some content and a state that governs its stying.
```

Comment:
You mention "its circle" before saying it can have a circle. Not sure exactly how to improve this but it's stylistically dubious. :-)